### ISSUE TITLE
Maximun heat temperature allowed is 25°

### ISSUE THREAD

**Issue by peibol81** [grzegorz914/homebridge-melcloud-control#83]
With 16.16 the maximum temperature allowed by the plugin in heat mode is 25°. In melcloud web and app is possible configure the unit to 26°, 27°...

**Comment by grzegorz914** [comment]
Which device?

**Comment by grzegorz914** [comment]
In ka case I can set to 31

**Comment by peibol81** [comment]
Mitsubishi air conditioning system (by grids).

**Comment by grzegorz914** [comment]
The home app allow to set between 10 and 31 for AC. Get the screenshot where you can’t set more as 25.

**Comment by peibol81** [comment]
In the homekit screen, when I select a temperature above 25, the temperature automatically changes to 25. automatically

**Comment by peibol81** [comment]
Today I updated nodejs to 20.11.1, maybe can be in relation?

**Comment by grzegorz914** [comment]
I don’t know but is definitely something wrong in your local environment, I have checked this and all tem can be set.

**Comment by peibol81** [comment]
Don't worry. In little ocasions I need more if 25°. Thank you for your time trying to help.

**Comment by peibol81** [comment]
Some news.
In 0.16.2 version, all works well.
In the next version, 0.16.13 is imposible set heat tempeeature above 25°. Siri says 'Value above the maximum allowed'

**Comment by grzegorz914** [comment]
I don’t have this behavior on my side, if so please post debug log during set tem above 25C

**Comment by peibol81** [comment]
Don't worry. I'm a blind person and this kind of things (paste debug info, etc...) is difficult because the app is not much accesible.
I will wait to future versions and if the problem continues I will talk to a friend to help me to paste the debug info.
Maybe is not a plugin problem. Don't worry, is not a fatal error.
Thank you again.

**Comment by ForkoB** [comment]
I also get an error
**This plugin generated a warning from the characteristic 'Heating Threshold Temperature': characteristic was supplied illegal value: number 27 exceeded maximum of 25. See https://homebridge.io/w/JtMGR for more info.**

**Comment by daniburton1** [comment]
Same error 
<img width="324" alt="image" src="https://github.com/grzegorz914/homebridge-melcloud-control/assets/160672437/4bcac50f-a13a-4a40-adce-870874793b0d">

**Comment by peibol81** [comment]
If any of you can put the plugin log here, it may be easier to find the problem. For me it is complicated because the process of generating and copying the log here is not accessible. I use Voiceover and although I've tried, I haven't succeeded.

However, I'm not sure it's a problem with the plugin. For me, this problem arises from the update to iOS 17.3.1. Before upgrading to that version, everything worked fine. I can also say that I have downgraded to the version of the 0.16.2 plugin and it works correctly again. That is the latest version of the plugin that works well for me, even with iOS 17.3.1. The bug may have been earlier but I realized a few days after installing that version of iOS and I don't know if it has to do with it. Be that as it may, for me it is not a serious failure because on very few occasions the heating temperature is 25°. I'm sorry I can't give more information. Thank you for the time and effort to try to solve the problems and improve the plugin. It makes my life easier.

**Comment by grzegorz914** [comment]
Please post the debug log, because I’m on latest plugin version and iOS 17.4 and the temp can be set to 31C. Have You other Heather/Cooler or Thermostat in Home app which can be set to 31C?

**Comment by daniburton1** [comment]
![image](https://github.com/grzegorz914/homebridge-melcloud-control/assets/160672437/1c21c7f9-9d61-4a48-9ea5-56258f991750)
![image](https://github.com/grzegorz914/homebridge-melcloud-control/assets/160672437/7ccc0008-a73b-4aaa-be41-c1e8cac5724d)

**Comment by daniburton1** [comment]
ON DEBUG MODE :

```
ock physical controls: UNLOCKED
[2/21/2024, 7:44:29 PM] [homebridge-melcloud-control] Air Conditioner Mansarda Temperature display unit: °C
[2/21/2024, 7:44:29 PM] [homebridge-melcloud-control] This plugin generated a warning from the characteristic 'Heating Threshold Temperature': characteristic was supplied illegal value: number 26 exceeded maximum of 25. See https://homebridge.io/w/JtMGR for more info.
[2/21/2024, 7:44:29 PM] [homebridge-melcloud-control] Error: 
    at HeatingThresholdTemperature.Characteristic.characteristicWarning (/homebridge/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:3011:105)
    at HeatingThresholdTemperature.Characteristic.validateUserInput (/homebridge/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:2927:14)
    at HeatingThresholdTemperature.<anonymous> (/homebridge/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Characteristic.ts:2405:24)
    at step (/homebridge/node_modules/homebridge/node_modules/tslib/tslib.js:195:27)
    at Object.next (/homebridge/node_modules/homebridge/node_modules/tslib/tslib.js:176:57)
    at fulfilled (/homebridge/node_modules/homebridge/node_modules/tslib/tslib.js:166:62)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Comment by daniburton1** [comment]
Update: 25°C limit seems to be applied only on heating, because on cooling and automode it can be highened to more then 30°C

**Comment by ForkoB** [comment]
> Please post the debug log, because I’m on latest plugin version and iOS 17.4 and the temp can be set to 31C. Have You other Heather/Cooler or Thermostat in Home app which can be set to 31C?

It is not connected with iOS 17.4 or 17.3.1
I deleted the bridge from the Homekit, restarted the Homebridge, and it returns the same error (without to be added back to the Homekit.

**Comment by grzegorz914** [comment]
Ok, I will look into this deeper next week because Im on holiday..

**Comment by grzegorz914** [comment]
Is the problem with all modes: Auto, Heat, Cool?

**Comment by daniburton1** [comment]
Seems only on heat for me

**Comment by peibol81** [comment]
For me too, the problem is only on heat mode.

Nice holidays!

**Comment by grzegorz914** [comment]
should be fixed in 0.16.17

**Comment by daniburton1** [comment]
Thank you very much 🔝
